File: app-java/src/main/java/com/danke/http/sample/ApiService.java
package com.danke.http.sample;

import io.reactivex.Observable;
import retrofit2.Response;
import retrofit2.http.*;

/**
 * @author danke
 * @date 2018/6/27
 */
public interface ApiService {

    @GET("users/{user}")
    Observable<Response<UserResponse>> getUser(@Path("user") String user);
}

File: app-kotlin/src/main/kotlin/com/danke/http/sample/ApiService.kt
package com.danke.http.sample

import io.reactivex.Observable
import retrofit2.Response
import retrofit2.http.*

/**
 * @author danke
 * @date 2018/6/27
 */
interface ApiService {
    @GET("users/{user}")
    fun getUser(@Path("user") user: String): Observable<Response<UserResponse>>
}